packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/Banner/banner.js

```diff
@@ -0,0 +1,312 @@
+        wrapperStyles.backgroundRepeat = backgroundRepeat
+            ? 'repeat'
+            : 'no-repeat';
+    }
+
+    if (appearance === 'poster') {
+        wrapperStyles.minHeight = minHeight;
+        overlayStyles.minHeight = minHeight;
+        overlayStyles.paddingTop = paddingTop;
+        overlayStyles.paddingRight = paddingRight;
+        overlayStyles.paddingBottom = paddingBottom;
+        overlayStyles.paddingLeft = paddingLeft;
+        contentStyles.width = '100%';
+    } else {
+        wrapperStyles.minHeight = minHeight;
+        wrapperStyles.paddingTop = paddingTop;
+        wrapperStyles.paddingRight = paddingRight;
+        wrapperStyles.paddingBottom = paddingBottom;
+        wrapperStyles.paddingLeft = paddingLeft;
+    }
+
+    const appearanceOverlayClasses = {
+        poster: classes.posterOverlay,
+        'collage-left': classes.collageLeftOverlay,
+        'collage-centered': classes.collageCenteredOverlay,
+        'collage-right': classes.collageRightOverlay
+    };
+    const appearanceOverlayHoverClasses = {
+        poster: classes.posterOverlayHover,
+        'collage-left': classes.collageLeftOverlayHover,
+        'collage-centered': classes.collageCenteredOverlayHover,
+        'collage-right': classes.collageRightOverlayHover
+    };
+
+    const typeToPriorityMapping = {
+        primary: 'high',
+        secondary: 'normal',
+        link: 'low'
+    };
+
+    let BannerButton;
+    if (showButton !== 'never') {
+        const buttonClass =
+            showButton === 'hover' ? classes.buttonHover : classes.button;
+
+        BannerButton = (
+            <div className={buttonClass}>
+                <Button
+                    priority={typeToPriorityMapping[buttonType]}
+                    type="button"
+                >
+                    {buttonText}
+                </Button>
+            </div>
+        );
+    }
+
+    const overlayClass =
+        showOverlay === 'hover' && !hovered
+            ? appearanceOverlayHoverClasses[appearance]
+            : appearanceOverlayClasses[appearance];
+
+    let BannerFragment = (
+        <div className={classes.wrapper} style={wrapperStyles}>
+            <div className={overlayClass} style={overlayStyles}>
+                <div
+                    className={classes.content}
+                    style={contentStyles}
+                    dangerouslySetInnerHTML={toHTML(content)}
+                />
+                {BannerButton}
+            </div>
+        </div>
+    );
+
+    if (typeof link === 'string') {
+        const linkProps = resolveLinkProps(link, linkType);
+        const LinkComponent = linkProps.to ? Link : 'a';
+        BannerFragment = (
+            <LinkComponent
+                className={classes.link}
+                {...linkProps}
+                {...(openInNewTab ? { target: '_blank' } : '')}
+                onDragStart={event => {
+                    event.preventDefault();
+                }}
```

Comment:
Avoid recreating event handlers on render, since they have to be reattached. Create and memoize the handler, then attach it (or don't) as necessary.

```jsx
const Banner = props => {
  // define it with `useCallback`
  const handleDragStart = useCallback(event => {
    event.preventDefault()
  }, [])

  return <LinkComponent onDragStart={handleDragStart} />
}

// or, since there are no dependencies, just define it outside of the component
const handleDragStart = event => event.preventDefault

const Banner = props => {
  return <LinkComponent onDragStart={handleDragStart} />
}
```

Reply:
Resolved: https://github.com/magento/pwa-studio/pull/1872/commits/51e840ee0742ea494f1d6b57c0b062c8fe83d1c1

Reply:
Sorry, there was a small bug in that suggestion (forgot to call `preventDefault`).

```es6
const handleDragStart = event => event.preventDefault()
```

Reply:
@jimbo and I'm the silly one for blindly copying it without checking 🤦‍♂ 
Resolved: https://github.com/magento/pwa-studio/pull/1872/commits/e982a8cb3d3237c94df8329d000fdb57f24b3613